File: python/examples/cp.2.py
#!/usr/bin/python
# Written by andrewt@cse.unsw.edu.au  for COMP2041
# Simple cp implementation reading file into a list

import sys
if len(sys.argv) != 3:
    print >>sys.stderr, "Usage: %s <infile> <outfile>" % sys.argv[0]
    sys.exit(1)
lines = open(sys.argv[1]).readlines()
open(sys.argv[2], 'w').writelines(lines)

File: perl/examples/cp.2.pl
#!/usr/bin/perl -w
# Written by andrewt@cse.unsw.edu.au for COMP2041
# Simple cp implementation reading entire file into array
# note that <> returns an array of lines
# in a list context (in a scalar context it returns a single line)

die "Usage: $0 <infile> <outfile>\n" if @ARGV != 2;

$infile = shift @ARGV;
$outfile = shift @ARGV;

open(IN, "<$infile") || die("Cannot open $infile: $!");
@lines = <IN>;
close(IN);

open(OUT, ">$outfile") || die("Cannot open $outfile: $!");
print OUT @lines;
close(OUT);

exit 0;
